### ISSUE TITLE
Crash in editor when I rename font

### ISSUE THREAD

**Issue by DrMoriarty** [godotengine/godot#28395]
**Godot version:**

Godot 3.1.1.rc1 (4764e1797)

**OS/device including version:**

MacOSX 10.13.6

**Issue description:**

I have several .tres resource files with DynamicFont saved in them. All of them are used in many game scenes. When I try to rename one file it processed without any errors. Then I try to rename another font file and godot crashes with this stack:

```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fff7686af5a _sigtramp + 26
[2] HashMap<StringName, MethodInfo, HashMapHasherDefault, HashMapComparatorDefault<StringName>, (unsigned char)3, (unsigned char)8>::get_element(StringName const&) const (in godot.osx.tools.64) (hash_map.h:194)
[3] Node* Object::cast_to<Node>(Object*) (in godot.osx.tools.64) (object.h:588)
[4] Variant::operator Node*() const (in godot.osx.tools.64) (variant.cpp:1763)
[5] VariantObjectClassChecker<Node*>::check(Variant const&) (in godot.osx.tools.64) (method_bind.h:143)
[6] MethodBind2<Dictionary, Node*>::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:1425)
[7] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[8] MessageQueue::_call_function(Object*, StringName const&, Variant const*, int, bool) (in godot.osx.tools.64) (message_queue.cpp:0)
[9] MessageQueue::flush() (in godot.osx.tools.64) (message_queue.cpp:305)
[10] SceneTree::iteration(float) (in godot.osx.tools.64) (scene_tree.cpp:477)
[11] Main::iteration() (in godot.osx.tools.64) (main.cpp:1862)
[12] OS_OSX::run() (in godot.osx.tools.64) (os_osx.mm:2654)
[13] main (in godot.osx.tools.64) (godot_main_osx.mm:100)
[14] 14  libdyld.dylib                       0x00007fff7655c015 start + 1
[15] 15  ???                                 0x0000000000000004 0x0 + 4
-- END OF BACKTRACE --
```

**Steps to reproduce:**

**Minimal reproduction project:**

**Comment by YeldhamDev** [comment]
Could you provide a minimal reproduction project?

**Comment by DrMoriarty** [comment]
Checking 3.1.2-devel (e802da4e8)
Crashlog:
```
OpenGL ES 3.0 Renderer: NVIDIA GeForce GT 1030 OpenGL Engine
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fff7686af5a _sigtramp + 26
[2] 2   ???                                 0x00007fa3505f1240 0x0 + 140339404804672
[3] Node* Object::cast_to<Node>(Object*) (in godot.osx.tools.64) (object.h:588)
[4] Variant::operator Node*() const (in godot.osx.tools.64) (variant.cpp:1763)
[5] VariantObjectClassChecker<Node*>::check(Variant const&) (in godot.osx.tools.64) (method_bind.h:143)
[6] MethodBind2<Dictionary, Node*>::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:1425)
[7] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[8] MessageQueue::_call_function(Object*, StringName const&, Variant const*, int, bool) (in godot.osx.tools.64) (message_queue.cpp:0)
[9] MessageQueue::flush() (in godot.osx.tools.64) (message_queue.cpp:305)
[10] SceneTree::iteration(float) (in godot.osx.tools.64) (scene_tree.cpp:477)
[11] Main::iteration() (in godot.osx.tools.64) (main.cpp:1862)
[12] OS_OSX::run() (in godot.osx.tools.64) (os_osx.mm:2654)
[13] main (in godot.osx.tools.64) (godot_main_osx.mm:100)
[14] 14  libdyld.dylib                       0x00007fff7655c015 start + 1
[15] 15  ???                                 0x0000000000000004 0x0 + 4
-- END OF BACKTRACE --
```

Minimal project: 
[Editor Crash.zip](https://github.com/godotengine/godot/files/3130372/Editor.Crash.zip)

**Comment by DrMoriarty** [comment]
Some times the same crashlog I have when I move png resource from one folder to another (#28528)

**Comment by DrMoriarty** [comment]
Probably fixed by #29459

**Comment by hpvb** [comment]
Thanks for testing!
